SocialCareCaseViewerApi.Tests/V1/Factories/EntityFactoryTests.cs

```diff
@@ -88,56 +64,44 @@ public void CanMapWorkerFromInfrastructureToDomainWithoutTeamDetails()
         [Test]
         public void CanMapWorkerFromInfrastructureToDomainWithTeamDetails()
         {
-            var email = _faker.Internet.Email();
-            var firstName = _faker.Name.FirstName();
-            var lastName = _faker.Name.LastName();
-            var id = 1;
-            var role = _faker.Random.Word();
-            int allocationCount = 1; //open allocations
-            var contextFlag = "A";
-            var createdBy = _faker.Internet.Email();
-            var dateStart = DateTime.Now;
-
-            var dbWorker = new dbWorker()
+            var workers = new List<dbWorker>
             {
-                Email = email,
-                FirstName = firstName,
-                LastName = lastName,
-                Id = id,
-                Role = role,
-                ContextFlag = contextFlag,
-                CreatedBy = createdBy,
-                DateStart = dateStart,
-                Allocations = new List<AllocationSet>() {
-                    new AllocationSet() { Id = 1, PersonId = 2, CaseStatus = "Closed" },
-                    new AllocationSet() { Id = 2, PersonId = 3, CaseStatus = "Open" }
-                },
-                WorkerTeams = new List<WorkerTeam>()
-                {
-                    new WorkerTeam() { Id = 1 , Team = new dbTeam() { Id = 1, Name = "Team 1", Context = "C" } },
-                    new WorkerTeam() { Id = 2 , Team = new dbTeam() { Id = 2, Name = "Team 2", Context = "C" } },
-                }
+                TestHelpers.CreateWorker(), TestHelpers.CreateWorker(hasAllocations: false, hasWorkerTeams: false)
             };
 
-            var expectedResponse = new Worker()
+            var expectedResponse = new List<Worker>
             {
-                FirstName = firstName,
-                LastName = lastName,
-                Id = id,
-                AllocationCount = allocationCount,
-                Email = email,
-                Role = role,
-                ContextFlag = contextFlag,
-                CreatedBy = createdBy,
-                DateStart = dateStart,
-                Teams = new List<Team>()
+                new Worker
+                {
+                    Id = workers[0].Id,
+                    Email = workers[0].Email,
+                    Role = workers[0].Role,
+                    ContextFlag = workers[0].ContextFlag,
+                    CreatedBy = workers[0].CreatedBy,
+                    FirstName = workers[0].FirstName,
+                    LastName = workers[0].LastName,
+                    DateStart = workers[0].DateStart,
+                    AllocationCount =
+                        workers[0].Allocations?.Count(allocation => allocation.CaseStatus.ToUpper() == "OPEN") ?? 0,
+                    Teams = workers[0].WorkerTeams?.Select(x =>
+                        new Team() {Id = x.Team.Id, Name = x.Team.Name, Context = x.Team.Context}).ToList() ?? new List<Team>()
```

Comment:
Possibly a silly question: Do we still need to do the null checks if the first worker will always have allocations and teams? Looks like the test helper will always create 3 of them.

Reply:
You're right that at runtime we will always have 3 WorkerTeams so therefore workers[0].WorkerTeams will never be null.

However, the compiler does not know if WorkerTeams will exist as it can be null, therefore at a minimum this deals with a compiler warning.

If something can be null I think it is good to handle null even when we are confident then it won't actually be null. 

This is just test code but NullReferenceExceptions have been the cause of probably most of our annoying backend bugs so far.

This is why I really like the use of #nullable as we get to be explicit about what can and can't be null then are forced to handle null correctly.

Reply:
Ahhh okay, that makes sense! Still remembering we're in C# world. 😝